test/e2e/e2e_test.go

```diff
@@ -108,68 +133,119 @@ func TestTaskRun(t *testing.T) {
+	t.Run("check deletion", func(t *testing.T) {
 		if err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (done bool, err error) {
-			tr, err := c.TaskRuns(ns).Get(ctx, tr.GetName(), metav1.GetOptions{})
-			if k8serrors.IsNotFound(err) {
-				return true, nil
+			_, err = tc.TaskRuns(defaultNamespace).Get(ctx, tr.GetName(), metav1.GetOptions{})
+			if err != nil {
+				if k8serrors.IsNotFound(err) {
+					return true, nil
+				}
+				t.Errorf("Error getting PipelineRun: %v", err)
+				return false, err
 			}
-			t.Logf("Get: %+v, %v", tr.GetName(), err)
 			return false, nil
 		}); err != nil {
-			t.Fatalf("error waiting TaskRun to be deleted: %v", err)
+			t.Fatalf("Error waiting for TaskRun deletion: %v", err)
+		}
+	})
+
+	t.Run("check result", func(t *testing.T) {
+		if resName == "" {
+			t.Skip("Result name not found")
+		}
+		_, err = gc.GetResult(context.Background(), &resultsv1alpha2.GetResultRequest{Name: resName})
+		if err != nil {
+			t.Errorf("Error getting Result: %v", err)
+		}
+	})
+
+	t.Run("check record", func(t *testing.T) {
+		if recName == "" {
+			t.Skip("Record name not found")
+		}
+		_, err = gc.GetRecord(context.Background(), &resultsv1alpha2.GetRecordRequest{Name: recName})
+		if err != nil {
+			t.Errorf("Error getting Record: %v", err)
 		}
 	})
 }
 
 func TestPipelineRun(t *testing.T) {
 	ctx := context.Background()
-	pr := new(v1beta1.PipelineRun)
-	b, err := ioutil.ReadFile("testdata/pipelinerun.yaml")
+	pr := new(tektonv1beta1.PipelineRun)
+	b, err := os.ReadFile("testdata/pipelinerun.yaml")
 	if err != nil {
-		t.Fatalf("ioutil.Readfile: %v", err)
+		t.Fatalf("Error reading file: %v", err)
 	}
 	if err := yaml.UnmarshalStrict(b, pr); err != nil {
-		t.Fatalf("yaml.Unmarshal: %v", err)
+		t.Fatalf("Error unmarshalling: %v", err)
 	}
 
-	c := clientTekton(t)
+	tc := tektonClient(t)
 
 	// Best effort delete existing Run in case one already exists.
-	_ = c.PipelineRuns(ns).Delete(ctx, pr.GetName(), metav1.DeleteOptions{})
+	_ = tc.PipelineRuns(defaultNamespace).Delete(ctx, pr.GetName(), metav1.DeleteOptions{})
 
-	if _, err = c.PipelineRuns(ns).Create(ctx, pr, metav1.CreateOptions{}); err != nil {
-		t.Fatalf("Create: %v", err)
+	if _, err = tc.PipelineRuns(defaultNamespace).Create(ctx, pr, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("Error creating PipelineRun: %v", err)
 	}
 
-	t.Run("result annotations", func(t *testing.T) {
+	gc, _ := resultsClient(t, allNamespacesReadAccessTokenFile, nil)
+
+	var resName, recName string
+
+	t.Run("check annotations", func(t *testing.T) {
 		// Wait for Result ID to show up.
 		if err := wait.PollImmediate(1*time.Second, 1*time.Minute, func() (done bool, err error) {
-			pr, err := c.PipelineRuns(ns).Get(ctx, pr.GetName(), metav1.GetOptions{})
+			pr, err := tc.PipelineRuns(defaultNamespace).Get(ctx, pr.GetName(), metav1.GetOptions{})
 			if err != nil {
-				t.Logf("Get: %v", err)
-				return false, nil
+				t.Fatalf("Error getting PipelineRun: %v", err)
```

Comment:
Same add `return false, err`

Reply:
Ditto from above - we can't return any value here, and `t.Fatalf` stops the test.

Reply:
This is the return statement of the callback function to `PollImmediate`. I guess calling `t.Fatal()` here would mark the test as failed before even waiting for the timeout.